### PR TITLE
Feature/new form in analysis edit

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

**Review by AdityaKhatri — APPROVED** [review]

File: src/views/Home/index.tsx [hunk 1]
```diff
@@ -238,13 +238,12 @@ function Home(props: ViewProps) {
                         headerActions={(
                             <>
                                 <SelectInput
+                                    name=""
```
Comment (tnagorra) [inline src/views/Home/index.tsx]:
```suggestion
                                    name={undefined}
```

File: src/components/viz/Timeline/index.tsx [hunk 2]
```diff
@@ -92,19 +93,26 @@ function Timeline<T>(props: TimelineProps<T>) {
         tickLabelSelector,
         data,
         domain: domainFromProps,
+        minDomainExtent = 7776000000, // 3 monts
```
Comment (tnagorra) [inline src/components/viz/Timeline/index.tsx]:
```suggestion
        minDomainExtent = 7776000000, // 3 months
```

File: src/constants/styleProperties.js [hunk 3]
```diff
@@ -66,10 +66,43 @@ const styleProperties = {
...
+        // spacingMedium: '0.79rem',
+        // spacingMediumAlt: '0.933rem',
+        // spacingLarge: '1.17rem',
+        // spacingLargeAlt: '1.36rem',
+        // spacingExtraLarge: '1.83rem',
+        // spacingExtraLargeAlt: '2.46rem',
+        // spacingSuperLarge: '3.01rem',
+        // spacingSuperLargeAlt: '3.81rem',
+        // spacingMegaLarge: '4.81rem',
+        // spacingUltraLarge: '6.89rem',
```
Comment (tnagorra) [inline src/constants/styleProperties.js]: Why add these comments? Please block comment if necessary to comment

File: src/views/AnalysisModule/AnalysisEditModal/index.tsx [hunk 4]
```diff
@@ -162,150 +165,172 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
...
-                >
-                    <NonFieldErrors
-                        persistent={false}
-                        faramElement
-                    />
+            {error?.$internal && (
+                <div className={styles.error}>
+                    { error?.$internal }
+                </div>
+            )}
```
Comment (tnagorra) [inline src/views/AnalysisModule/AnalysisEditModal/index.tsx]: This is dangerous!

File: src/views/AnalysisModule/AnalysisEditModal/index.tsx [hunk 5]
```diff
@@ -162,150 +165,172 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
...
+                error={error?.fields?.teamLead}
+                onChange={onValueChange}
+                disabled={pending}
+            />
+            <div className={styles.analysisPillarListContainer}>
+                {apInternalError && (
+                    <div>
+                        {apInternalError}
+                    </div>
+                )}
```
Comment (tnagorra) [inline src/views/AnalysisModule/AnalysisEditModal/index.tsx]: This may be dangerous

File: src/views/AnalysisModule/AnalysisEditModal/index.tsx [hunk 6]
```diff
@@ -162,150 +165,172 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
...
+                <div className={styles.analysisPillarList}>
                     <List
-                        faramElement
+                        data={value.analysisPillar ?? []}
```
Comment (tnagorra) [inline src/views/AnalysisModule/AnalysisEditModal/index.tsx]: List should handle this

File: src/views/Home/Assignment/index.tsx [hunk 7]
```diff
@@ -133,13 +134,23 @@ function Assignments() {
...
+                    />
+                ) : (
+                    <div className={styles.emptyMessage}>
+                        <MdPlaylistAddCheck className={styles.icon} />
+                        <div className={styles.text}>
+                            {/* FIXME: use strings with appropriate wording */}
+                            You do not have any assignments
+                        </div>
+                    </div>
+                )}
```
Comment (tnagorra) [inline src/views/Home/Assignment/index.tsx]: Shouldn't we also check for loading state?

File: src/views/Home/RecentActivity/index.tsx [hunk 8]
```diff
@@ -52,12 +51,22 @@ function RecentActivities() {
...
+                    />
+                ) : (
+                    <div className={styles.emptyMessage}>
+                        <GiShrug className={styles.icon} />
+                        <div className={styles.text}>
+                            {/* FIXME: use strings with appropriate wording */}
+                            You do not have any recent activities
+                        </div>
+                    </div>
+                )}
```
Comment (tnagorra) [inline src/views/Home/RecentActivity/index.tsx]: Shouldn't we also check for loading state.

Maybe introduce props:
- loading
- emptyElement
- loadingElement

inside list?

File: src/views/Home/index.tsx [hunk 9]
```diff
@@ -257,12 +254,24 @@ function Home(props: ViewProps) {
...
+                            <div className={styles.emptyRecentProject}>
+                                <GiShrug className={styles.icon} />
+                                <div className={styles.text}>
+                                    {/* FIXME: use strings with appropriate wording */}
+                                    Looks like you do not have any recent project,
+                                    <br />
+                                    please select a project to view it&apos;s details
+                                </div>
+                            </div>
+                        )}
```
Comment (tnagorra) [inline src/views/Home/index.tsx]: What about loading state or errored state?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by samshara — APPROVED** [review]

**Pull request closed without merging**